lib/API/JsonApi/v1/Filters/Interpreters/FilterInterpreter.php

```diff
@@ -0,0 +1,332 @@
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\API\JsonApi\v1\Filters\Interpreters;
+
+use SuiteCRM\API\JsonApi\v1\Filters\Interfaces\OperatorInterface;
+use SuiteCRM\API\JsonApi\v1\Filters\Interpreters\ByIdFilters\ByIdFilter;
+use SuiteCRM\API\JsonApi\v1\Filters\Operators\FieldOperator;
+use SuiteCRM\API\JsonApi\v1\Filters\Operators\Operator;
+use SuiteCRM\API\JsonApi\v1\Filters\Validators\FieldValidator;
+use SuiteCRM\API\v8\Exception\BadRequest;
+use SuiteCRM\Exception\Exception;
+
+use Psr\Container\ContainerInterface;
+
+class FilterInterpreter
+{
+    /**
+     * @var ContainerInterface $containers
+     */
+    private $containers;
+    private static $fieldOperator;
+    private static $filterOperators;
+    private static $filterFieldOperators;
+    private static $filterSpecialOperators;
+
+    /**
+     * FilterInterpreter constructor.
+     * @param ContainerInterface $containers
+     * @throws \Psr\Container\ContainerExceptionInterface
+     */
+    public function __construct(ContainerInterface $containers)
+    {
+        $this->containers = $containers;
+
+        if (self::$fieldOperator === null) {
+            self::$fieldOperator = new FieldOperator($containers);
+        }
+
+        if (self::$filterOperators === null) {
+            self::$filterOperators = $containers->get('FilterOperators');
+        }
+
+        if (self::$filterFieldOperators === null) {
+            self::$filterFieldOperators = $containers->get('FilterFieldOperators');
+        }
+
+        if (self::$filterSpecialOperators === null) {
+            self::$filterSpecialOperators = $containers->get('FilterSpecialOperators');
+        }
+    }
+
+    /**
+     * @param $filterStructure
+     * @return bool
+     * @throws Exception
+     */
+    public function isFilterByPreMadeName($filterStructure) {
+        if(is_array($filterStructure) === false) {
+            throw new Exception('[JsonApi][v1][Filters][Interpreters][isFilterByPreMadeName][expected type to be array]');
+        }
+
+        return count($filterStructure) === 1 && is_array(current($filterStructure)) === false;
+    }
+
+    /**
+     * @param $filterStructure
+     * @return bool
+     * @throws Exception
+     */
+    public function isFilterById($filterStructure) {
+        if(is_array($filterStructure) === false) {
+            throw new Exception('[JsonApi][v1][Filters][Interpreters][isFilterById][expected type to be array]');
+        }
+
+        return count($filterStructure) === 1 &&
+            array_keys($filterStructure)[0] === '[id]' &&
+            is_array(current($filterStructure)) === true;
+    }
+
+    /**
+     * @param $filterStructure
+     * @return bool
+     * @throws Exception
+     */
+    public function isFilterByAttributes($filterStructure) {
+        if(is_array($filterStructure) === false) {
```

Comment:
is_array's return values are true or false....

Reply:
?

Reply:
You can use simply !is_array here

Reply:
The criteria for a code review is to check that it is functional, maintainable and readable. This code meets this criteria. I choose this style, because I wish to stress that that this value needs to be false.

Reply:
No, it doesn't. You make an unnecessarry comparison.